geode-core/src/main/java/org/apache/geode/internal/cache/wan/GatewaySenderEventImpl.java

```diff
@@ -754,8 +754,8 @@ protected void serializeKey(DataOutput out,
   @Override
   public void fromData(DataInput in,
       DeserializationContext context) throws IOException, ClassNotFoundException {
-    fromDataPre_GEODE_1_13_0_0(in, context);
-    if (version >= KnownVersion.GEODE_1_13_0.ordinal()) {
+    fromDataPre_GEODE_1_14_0_0(in, context);
+    if (version >= KnownVersion.GEODE_1_14_0.ordinal()) {
```

Comment:
There is no need for condition check here. The appropriate version of `fromData()` is invoked by `DSFIDSerializerImpl.invokeFromData()` and `InternalDataSerializer.invokeFromData()`.  For older versions, it will not invoke `fromData()`. Instead, it will invoke `fromDataPre_GEODE_1_9_0_0` or `fromDataPre_GEODE_1_14_0_0` depending on the version from `DataInput` or `DeserializationContext`.